marathon_lb.py

```diff
@@ -700,27 +732,28 @@ def reloadConfig():
 
 def generateHttpVhostAcl(
         templater, app, backend, haproxy_map, map_array,
-        haproxy_dir, duplicate_map):
+        haproxy_dir, duplicate_map, https_grouped_frontend_list):
```

Comment:
is this new param an "output" param? it looks like this func mutates its value, which seems surprising. is that an idiom that we want to encourage in this codebase?

Reply:
It isn't a pure function because this function also mutates a duplicate_map.
But I agree with you and will fix that